src/nxdoc/nuxeo-server/data-store/indexing-and-query/moving-load-from-database-to-elasticsearch.md

```diff
@@ -138,17 +139,18 @@ Can be done like this:
 doc.setPropertyValue("dc:title", "A new title");
 session.saveDocument(doc);
 session.save();
-
-ElasticSearchAdmin esa = Framework.getService(ElasticSearchAdmin.class);
+// flag the thread to wait for syncrhonous indexing on commit 
+ThreadLocalIndexingCommandsStacker.useSyncIndexing.set(true);
+// commit
 TransactionHelper.commitOrRollbackTransaction();
 TransactionHelper.startTransaction();
-esa.prepareWaitForIndexing().get(20, TimeUnit.SECONDS); // wait for indexing
-esa.refresh(); // explicit refresh
 
-ess.query(new NxQueryBuilder(session).nxql("SELECT * FROM Document WHERE dc:title = 'A new title'")); // "doc" is returned
+SearchService searchService = Framework.getService(SearchService.class);
+SearchResponse response = searchService.search(SearchQuery.Builder("SELECT * FROM Document WHERE dc:title = 'A new title'").build());
+DocumentModelList docs = response.loadDocuments(session); // "doc" is returned
 ```
 
-Obviously there is a write overhead here because we are splitting the transaction and explicitly call a refresh. This can be useful for unit test migration but on normal code you have to decide if it make sense to search documents that are probably already loaded in your context.
+Obviously there is a write overhead here because we are splitting the transaction and explicitly call a refresh. This can be useful but on normal code you have to decide if it makes sense to search documents that are probably already loaded in your context. Note that on unit test you don't have to do this because the `CoreSearchFeature` is already configured to wait for syncrhonous indexing during commit. 
```

Comment:
are we still calling refresh?
also: "in unit tests", "synchronous indexing"

Reply:
yes there is a refresh call done by the synchronous indexer.